core/pkg/resourcehandler/remotegitutils.go

```diff
@@ -18,9 +38,26 @@ func cloneRepo(gitURL giturl.IGitURL) (string, error) {
 		return "", fmt.Errorf("failed to create temporary directory: %w", err)
 	}
 
-	// Clone option
+	// Get the URL to clone
 	cloneURL := gitURL.GetHttpCloneURL()
-	cloneOpts := git.CloneOptions{URL: cloneURL}
+
+	isGitRepoPublic := isGitRepoPublic(cloneURL)
+
+	// Declare the authentication variable required for cloneOptions
+	var auth transport.AuthMethod
+
+	if isGitRepoPublic {
+		// No authentication needed if repository is public
+		auth = nil
+	} else {
+		auth = &http.BasicAuth{
```

Comment:
We should check if there is a token- if not, the function should return an error